haystack/evaluation/eval_run_result.py

```diff
@@ -119,7 +119,7 @@ def comparative_individual_scores_report(self, other: "BaseEvaluationRunResult")
             this_name = f"{this_name}_first"
             other_name = f"{other_name}_second"
 
-        if self.inputs != other.inputs:
+        if self.inputs.keys() != other.inputs.keys():
```

Comment:
Why was this changed? The original code was intentionally comparing both keys and values (as it only sense to compare results of eval runs that have the same input).

Reply:
We should only compare the keys since:

```
dict_keys(['question', 'contexts', 'answer', 'predicted_answer'])
```

If we compare the values, `contexts` and `predicted_answer` will mos probably be different.

Reply:
Ah, fair point. I still think the queries should be the same between different runs for the comparison to make sense , but that's specific to the RAG usecase and shouldn't be part of the generic results struct. 

Let's do the following - we change the comparison to just the keys and modify the warning to reflect that. However, it should happen in a different PR (this one isn't dependent on the change).

Reply:
@shadeMe https://github.com/deepset-ai/haystack/pull/7603